pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -706,4 +714,67 @@ public void close() {
             logger.warn("Failed to close producers", e);
         }
     }
+
+    @Override
+    public void seek(String topic, int partition, MessageId messageId) throws PulsarClientException {
+        Consumer<?> consumer = getConsumer(topic, partition);
+        consumer.seek(messageId);
+    }
+
+    @Override
+    public void pause(String topic, int partition) throws PulsarClientException {
+        getConsumer(topic, partition).pause();
+    }
+
+    @Override
+    public void resume(String topic, int partition) throws PulsarClientException {
+        getConsumer(topic, partition).resume();
+    }
+
+    public void setInputConsumers(List<Consumer<?>> inputConsumers) {
+        this.inputConsumers = inputConsumers;
+        inputConsumers.stream()
+            .flatMap(consumer ->
+                    consumer instanceof MultiTopicsConsumerImpl
+                            ? ((MultiTopicsConsumerImpl<?>) consumer).getConsumers().stream()
+                            : Stream.of(consumer))
+            .forEach(consumer -> topicConsumers.putIfAbsent(TopicName.get(consumer.getTopic()), consumer));
+    }
+
+    @VisibleForTesting
+    Consumer<?> getConsumer(String topic, int partition) throws PulsarClientException {
+        if (inputConsumers == null) {
+            throw new PulsarClientException("Getting consumer is not supported");
+        }
+        for (int i = 0; i < 2; i++) {
+            Consumer<?> consumer = topicConsumers.get(TopicName.get(topic).getPartition(partition));
+
+            if (consumer != null) {
+                return consumer;
+            }
+
+            if (partition == 0) {
+                consumer = topicConsumers.get(TopicName.get(topic));
+
+                if (consumer != null) {
+                    return consumer;
+                }
+            }
+
+            if (i == 0) {
```

Comment:
I am not sure I understand why in this "get" method we are modifying the internal `inputConsumers` variable.

Reply:
Try to get a consumer. If not found,  reprocess MultiTopicsConsumers in case new consumers appeared (happens on repartition or a new topic that matches provided pattern if the pattern is used), give it another try. 
it modifies `topicConsumers` (map of topic to a consumer.)

Reply:
I see